server/native/gamestate/src/game.rs

```diff
@@ -1093,202 +1080,101 @@ impl GameState {
-                },
-            );
-
-            if player.character.name == Name::H4ck {
-                match player.effects.get(&Effect::NeonCrashing) {
-                    Some(EffectData {
-                        direction: Some(direction),
-                        ..
-                    }) => {
-                        let speed = player.speed() as i64;
-                        GameState::move_player_to_direction(
-                            &mut self.board,
-                            &mut player.position,
-                            direction,
-                            speed,
-                        )
-                        .unwrap();
-
-                        let damage_neon_crash = player.skill_3_damage() as i64;
-                        let range_neon_crash = player.skill_3_range() as f64;
-
-                        let af_pl = GameState::affected_players(
-                            range_neon_crash,
-                            &pys,
-                            &player.position,
-                            player.id,
-                        );
-                        neon_crash_affected_players
-                            .insert(player.id, (damage_neon_crash, af_pl, None));
-                    }
-                    _ => {}
-                }
-            }
-
-            if player.character.name == Name::Muflus {
-                match player.effects.get(&Effect::Leaping) {
-                    Some(EffectData {
-                        direction: Some(direction),
-                        ..
-                    }) => {
-                        let speed = player.speed() as i64;
-                        GameState::move_player_to_direction(
-                            &mut self.board,
-                            &mut player.position,
-                            direction,
-                            speed,
-                        )
-                        .unwrap();
-                    }
-                    _ => {}
-                }
-            }
-        });
+    /// Move with a dash movement, mostly used on world tick.
+    pub fn move_with_dash(
+        player: MutablePlayer,
+        dashing_player_speed: u64,
+        dashing_player_id: u64,
+        players: &MutablePlayers,
+        board: &Board,
+        affected_players: &mut HashMap<u64, (i64, Vec<u64>)>,
+        direction: &RelativePosition,
+    ) -> Result<(), String> {
+        let dashing_player_position = player.position();
+        let new_position = GameState::move_player_to_direction(
+            &board,
+            &dashing_player_position,
+            direction,
+            dashing_player_speed as i64,
+        )?;
+        player.set_position(new_position);
+        *affected_players = GameState::affected_players(
+            2,
+            200.,
+            // TODO: Remove this if we ever use MutablePlayers
+            (players.clone()).into_iter().map(Into::into).collect(),
```

Comment:
I'm not sure you need to clone the vector here. You could do this instead:
```suggestion
            players.iter().map(Into::into).collect(),
```
Or if that doesn't work:
```suggestion
            players.iter().cloned().map(Into::into).collect(),
```

Reply:
Hey @Oppen 
Thanks for the comment! I'll try that and let you know 👍